### PR TITLE
Capture and record output from tests from Console.Out into the test results

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This helps bring parity to diagnosing test failures when using a merged test runner while not blocking console output.

### CODE REVIEW

File: src/tests/Common/XUnitWrapperLibrary/TestSummary.cs [hunk 1]
```diff
@@ -72,15 +72,15 @@ public string GetTestResultOutput(string assemblyName)
...
+                resultsFile.AppendLine($@"result=""Fail""><failure exception-type=""{test.Exception.GetType()}""><message><![CDATA[{test.Exception.Message}]]></message><stack-trace><![CDATA[{test.Exception.StackTrace}]]></stack-trace></failure><output><![CDATA[{test.Output}]]></output></test>");
             }
             else if (test.SkipReason is not null)
             {
                 resultsFile.AppendLine($@"result=""Skip""><reason><![CDATA[{test.SkipReason}]]></reason></test>");
             }
             else
             {
-                resultsFile.AppendLine(@" result=""Pass"" />");
+                resultsFile.AppendLine($@" result=""Pass""><output><![CDATA[{test.Output}]]></output></test>");
```
Comment (trylek) [inline src/tests/Common/XUnitWrapperLibrary/TestSummary.cs:83]: I believe today xUnit doesn't store output from successful tests in the xml summary. In general I like the idea for better telemetry but we should try to make sure we're not ending up with gigabytes of xml results. Could you please locally try out the merged TypeGeneratorTests to double-check the lengths of the xml summaries they produce? These tests are especially chatty and together they amount for about 1/7 of the entire Pri1 test set so hopefully the reading should give us a reasonable estimate of the xml sizes.

**Review by trylek — APPROVED** [review]
Looks good to me modulo one question, thank you!

File: src/tests/Common/XUnitWrapperLibrary/TestSummary.cs (hunk 1 above)
Comment (jkoritzinsky) [reply src/tests/Common/XUnitWrapperLibrary/TestSummary.cs:83]: With the whole TypeGeneratorTests suite `0-1600`, the total size of all the logs comes out to 154MB when we include successful output in the log. Is that low enough for us to keep the output in the log, or should we change it to only include it in failure cases?

Comment (trylek) [reply src/tests/Common/XUnitWrapperLibrary/TestSummary.cs:83]: Hmm, that probably means that the extrapolated size for the entire Pri1 test set is around 1 GB. I don't think that's a huge deal and for PR Pri0 sets it should amount to just 200 MB that's easily dwarfed by many of the build artifacts and binlogs. My suggestions is keep it for now - if we break the lab with the excessive outputs, Chris will let us know on Thursday and we'll just scale it back I guess ;-).

Comment (jkoritzinsky) [reply src/tests/Common/XUnitWrapperLibrary/TestSummary.cs:83]: Sounds good to me

**Pull request merged**